metricbeat/module/docker/cpu/helper.go

```diff
@@ -80,27 +80,30 @@ func getNewCpu(stats *dc.Stats) CPURaw {
 
 func (c *CPUService) PerCpuUsage(stats *dc.Stats) common.MapStr {
 	var output common.MapStr
-	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOLdCpu(stats).PerCpuUsage) {
+	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOldCpu(stats).PerCpuUsage) {
```

Comment:
> But the types i'm comparing are both arrays ( uint64 arrays ) & not slices!

No, they are slices. An array would be declared as `[n]type` where as a slice is `[]type`.

Here's a test and demonstrates a panic.

```
func TestPerCpuUsagePanic(t *testing.T) {
	stats := &dc.Stats{}
	stats.PreCPUStats.CPUUsage.PercpuUsage = make([]uint64, 1, 2)
	stats.CPUStats.CPUUsage.PercpuUsage = make([]uint64, 2)

	NewCpuService().PerCpuUsage(stats) // Panic
}
```